hikari/impl/bot.py

```diff
@@ -195,9 +195,9 @@ class GatewayBot(traits.GatewayBotAware):
         will __force__ colour to be used in console-based output. Specifying a
         `"CLICOLOR_FORCE"` environment variable with a non-`"0"` value will
         override this setting.
-    cache_settings : typing.Optional[hikari.config.CacheSettings]
+    cache_settings : typing.Optional[hikari.impl.config.CacheSettings]
```

Comment:
Shouldnt this be `api`?

Reply:
Nah, the idea here is that `hikari.impl.config` is the configuration objects the standard implementations take in and `hikari.api.config` are the fields that are guaranteed to be exposed on config properties between impls.

For this feature I took the liberty of breaking the normal standard of only ever exposing the api classes on return values as it is helpful for fields like `hikari.impl.GatewayBot.http_settings` to expose the config impl detail since this impl detail has a lot 
 more info and is guaranteed to stay the same for that class until a breaking change is made like switching to a different rest client.

Reply:
Although i do need to change `from hikari.impl import config` to `from hikari.impl import config as config_impl` in this module to make it more clear lol

Reply:
Fixed those inconsistent imports btw